src/Tasks.js

```diff
@@ -0,0 +1,496 @@
+}
+
+function TaskTitle({title,handleTitleChange}) {
+    const [taskTitle,setTaskTitle] = useState(title);
+
+    function handleChange(e) {
+        setTaskTitle(e.target.value);
+        handleTitleChange(e.target.value);
+    }
+
+    return (
+        <div className="taskTitleOverlay">
+            <label htmlFor="taskTitleInput"><i className="fa fa-pencil-square-o" aria-hidden="true"></i> Title :- </label>
+            <input value={taskTitle} onChange={handleChange} type="text" id="taskTitleInput" name="taskTitleInput" />  
+        </div>
+    )
+}
+
+function TaskAssignee({users,assignee,handleAssigneeChange}) {
+
+    const [taskAssignee,setTaskAssignee] = useState(assignee);
+
+    function handleChange(e) {
+        setTaskAssignee(e.target.value);
+        handleAssigneeChange(e.target.value);
+    }
+
+    return (
+        <div className="taskUserOverlay">
+            <label htmlFor="taskUserInput"><i className="fa fa-user-circle-o" aria-hidden="true"></i> Assigee :- </label>
+            <select id="taskUserInput" value={taskAssignee} onChange={handleChange}>
+                {users.map(function(user) {
+                    return (<option value={user.name} key={user.id}>{user.name}</option>);
+                })}
+            </select>  
+        </div>
+    )
+}
+
+function TaskStatus({status,handleStatusChange}) {
+
+    const [taskStatus,setTaskStatus] = useState(status);
+
+    function handleChange(e) {
+        setTaskStatus(e.target.value);
+        handleStatusChange(e.target.value);
+    }
+    return (
+        <div className="taskStatusOverlay">
+            <label htmlFor="taskSatusInput"><i className="fa fa-arrow-circle-right" aria-hidden="true"></i>
+                {` Status :-`}</label>
+            <select id="taskStatusInput" value={taskStatus} onChange={handleChange}>
+                <option value="Not started">Not started</option>
+                <option value="In-progress">In-progress</option>
+                <option value="On hold">On hold</option>
+                <option value="Completed">Completed</option>
+            </select>
+        </div>
+    );
+
+}
+
+function TaskDate({date,handleDateChange}) {
+    const [taskDate,setTaskDate] = useState(date);
+
+    function handleChange(e) {
+        setTaskDate(e.target.value);
+        handleDateChange(e.target.value);
+    }
+
+    return (
+        <div className="taskDateOverlay">
+            <label htmlFor="taskDateInput"><i className="fa fa-clock-o" aria-hidden="true"></i> Due date :- </label>
+            <input value={taskDate} onChange={handleChange} type="date" id="taskDateInput" name="taskDateInput" placeholder="dd/mm/yyyy" />  
+        </div>
+    )
+}
+
+
+function TaskOverlay({task,users,handleClose,handleSave,handleDelete}) {
+
+    let selectedIndex = useMemo(() => users.findIndex((user) => user.id == task.assignee),[task]);
+    
+    let taskImage = task.imageUrl || "assets/default.png";
+    function handleImageChange(imageUrl) {
```

Comment:
here, we are creating handlers for each field.

Can't we can keep a single `onAction` and use that?